lang/cs/Org.Apache.REEF.Client/Common/DriverFolderPreparationHelper.cs

```diff
@@ -112,25 +122,39 @@ internal void CreateDriverConfiguration(IJobSubmission jobSubmission, string dri
         /// <summary>
         /// Creates the driver folder structure in this given folder as the root
         /// </summary>
-        /// <param name="driverFolderPath"></param>
-        internal void CreateDefaultFolderStructure(string driverFolderPath)
+        /// <param name="jobSubmission">Job submission information</param>
+        /// <param name="driverFolderPath">Driver folder path</param>
+        internal void CreateDefaultFolderStructure(IJobSubmission jobSubmission, string driverFolderPath)
         {
             Directory.CreateDirectory(Path.Combine(driverFolderPath, _fileNames.GetReefFolderName()));
             Directory.CreateDirectory(Path.Combine(driverFolderPath, _fileNames.GetLocalFolderPath()));
             Directory.CreateDirectory(Path.Combine(driverFolderPath, _fileNames.GetGlobalFolderPath()));
-        }
 
-        /// <summary>
-        /// Adds all Assemlies to the Global folder in the Driver.
-        /// </summary>
-        private void AddAssemblies()
-        {
             var resourceHelper = new ResourceHelper(typeof(DriverFolderPreparationHelper).Assembly);
             foreach (var fileResources in clientFileResources)
             {
                 File.WriteAllBytes(resourceHelper.GetString(fileResources.Item1), resourceHelper.GetBytes(fileResources.Item2));
             }
+            var exePathLocal = Path.Combine(driverFolderPath, BridgeExe);
+            var exePathYarn = Path.Combine(driverFolderPath, _fileNames.GetReefFolderName(), BridgeExe);
+            var configPathLocal = Path.Combine(driverFolderPath, BridgeExeConfig);
+            var configPathYarn = Path.Combine(driverFolderPath, _fileNames.GetReefFolderName(), BridgeExeConfig);
+            File.Copy(BridgeExe, exePathLocal);
+            File.Copy(BridgeExe, exePathYarn);
+            var config = DefaultDriverConfigurationFileContents;
+            if (!string.IsNullOrEmpty(jobSubmission.DriverConfigurationFileContents))
+            {
+                config = jobSubmission.DriverConfigurationFileContents;
+            }
```

Comment:
what happens if user provided string does not include "probing privatePath=foo"?